packages/workbox-google-analytics/test/sw/index.js

```diff
@@ -0,0 +1,349 @@
+/*
+ Copyright 2017 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import IDBHelper from '../../../../lib/idb-helper.js';
+import * as googleAnalytics from '../../src/index.js';
+import constants from '../../src/lib/constants.js';
+import {QueuePlugin} from '../../../workbox-background-sync/src/index.js';
+import {NetworkFirst, NetworkOnly}
+    from '../../../workbox-runtime-caching/src/index.js';
+
+const PAYLOAD = 'v=1&t=pageview&tid=UA-12345-1&cid=1&dp=%2F';
+
+const sleep = (time) => new Promise((resolve) => setTimeout(resolve, time));
+
+const waitUntil = async (test) => {
+  if (test() === true) {
+    return Promise.resolve();
+  } else {
+    await sleep(100);
+    return waitUntil(test);
+  }
+};
+
+describe(`initialize`, function() {
+  let idbMap = {};
+
+  const stubIdbHelper = () => {
```

Comment:
Why are you stubbing all of this out?

Reply:
We discussed this briefly the other day, but I was having issues with multiple open connections to this IDB database at the same time, and that was preventing me from deleting the DB between tests.

I should confirm that this is still an issue if I delete all the records rather than deleting the DB itself.

Reply:
comments in the file other wise this will be totally forgotten and cause someone headaches later on.